diff --git a/test/Makefile b/test/Makefile
index bfe42c3..2c7e475 100644
--- a/test/Makefile
+++ b/test/Makefile
@@ -150,6 +150,13 @@ TESTFILES += ../testqsinglelsi.db
 TESTFILES += ../testioc.acf
 TESTS += testqsingle
 
+TESTPROD_HOST += testalarmmsg
+testalarmmsg_SRCS += testalarmmsg
+testalarmmsg_SRCS += testioc_registerRecordDeviceDriver.cpp
+testalarmmsg_LIBS = pvxsIoc pvxs $(EPICS_BASE_IOC_LIBS)
+TESTFILES += ../testalarmmsg.db
+TESTS += testalarmmsg
+
 endif
 
 ifdef BASE_7_0
