diff --git a/ioc/iocsource.cpp b/ioc/iocsource.cpp
index da2e251..5079233 100644
--- a/ioc/iocsource.cpp
+++ b/ioc/iocsource.cpp
@@ -19,6 +19,7 @@
 #include <pvxs/log.h>
 
 #include "alarm.h"
+#include "sitehooks.h"
 #include "iocsource.h"
 #include "dbentry.h"
 #include "dberrormessage.h"
@@ -349,6 +350,8 @@ void IOCSource::get(Value& node, // node within top level structure addressed by
             getArrayValue(pChannel, pDbFieldLog, value);
         }
     }
+
+    site::postProcessNode(dbChannelRecord(pChannel), node);
 }
 
 /**
